apps/nestjs-wallet/src/vc-api/vp-request-submission-verifier.ts

```diff
@@ -0,0 +1,100 @@
+import { IPresentation, IPresentationDefinition, PEX, ProofPurpose } from '@sphereon/pex';
+import { CredentialVerifier } from './credentials/types/credential-verifier';
+import { PresentationVerifier } from './credentials/types/presentation-verifier';
+import { VerifiablePresentation } from './credentials/types/verifiable-presentation';
+import { VpRequestEntity } from './exchanges/entities/vp-request.entity'; // Todo: have an interface instead of an entity
+import { VpRequestQueryType } from './exchanges/types/vp-request-query-type';
+
+/**
+ * Inspired by https://github.com/gataca-io/vui-core/blob/6c599bdf7086f9a702e6657089fa343ae62a417a/service/validatorServiceDIFPE.go
+ * Verifies:
+ *  - Signatures/Proofs
+ *  - Conformance with VpRequest (e.g. credential queries)
+ *  - Authority of Issuer: TODO TODO TODO
```

Comment:
@nichonien I'm hoping to use the role governance verification code that you are writing to do this bit